specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json

```diff
@@ -3439,28 +3594,57 @@
     "ReservationTransaction": {
       "description": "Reservation transaction resource.",
       "type": "object",
+      "discriminator": "kind",
       "allOf": [
         {
           "$ref": "#/definitions/Resource"
         }
       ],
       "properties": {
-        "properties": {
-          "x-ms-client-flatten": true,
-          "$ref": "#/definitions/ReservationTransactionProperties",
-          "title": "Reservation Transaction properties"
+        "kind": {
+          "type": "string",
+          "description": "Specifies the kind of reservation transaction.",
+          "enum": [
+            "Legacy",
+            "Modern"
```

Comment:
when does a legacy transaction get returned and when does a modern transaction get returned?

Reply:
When the API is called with a Modern billing account, a modern transaction is returned. When it is called with a legacy Enrollment number, legacy reservation transaction is returned.